### PR TITLE
refactor(ext/websocket): use fastwebsockets client

### PR DESCRIPTION
**Pull request by littledivy**
Depends on https://github.com/littledivy/fastwebsockets/pull/10

### CODE REVIEW

File: ext/websocket/lib.rs [hunk 1]
```diff
@@ -21,41 +17,39 @@ use deno_core::Resource;
...
-mod server;
+use fastwebsockets::CloseCode;
+use fastwebsockets::FragmentCollector;
+use fastwebsockets::Frame;
+use fastwebsockets::OpCode;
+use fastwebsockets::Role;
+use fastwebsockets::WebSocket;
 
-pub use server::ws_create_server_stream;
+pub use tokio_tungstenite; // Re-export tokio_tungstenite
```
Comment (bartlomieju) [inline ext/websocket/lib.rs:52]: Drop it?

File: ext/websocket/01_websocket.js [hunk 2]
```diff
@@ -508,7 +506,7 @@ class WebSocket extends EventTarget {
           if (prevState === OPEN) {
             try {
               await core.opAsync(
-                this[_role] === SERVER ? "op_server_ws_close" : "op_ws_close",
+                "op_ws_close",
```
Comment (bartlomieju) [inline ext/websocket/01_websocket.js:509]: Why are you going back to having a single op for both client and server?

**Review by bartlomieju — COMMENTED** [review]
Great to see so many lines removed. The implementation is dead simple now.

File: ext/websocket/01_websocket.js (hunk 2 above)
Comment (littledivy) [reply ext/websocket/01_websocket.js:509]: Because the client and server apis are the same with `fastwebsockets::Role` being `Client` or `Server`

**Review by bartlomieju — APPROVED** [review]
LGTM

**Review by bartlomieju — CHANGES_REQUESTED** [review]
Uh oh, WPT are failing

File: ext/websocket/02_websocketstream.js [hunk 3]
```diff
@@ -176,7 +176,7 @@ class WebSocketStream {
                         create.rid,
                       );
 
-                      if (kind > 6) {
```
Comment (bartlomieju) [inline ext/websocket/02_websocketstream.js:179]: Is there a test that covers this change? If not can you add it?

Comment (littledivy) [reply ext/websocket/02_websocketstream.js:179]: WPT covers it but this was never hit. I actually found this case by accident with a bug in fastwebsockets.

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**